sabnzbd/utils/correct_extension.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python3
+
+""" function to check and find correct extension of a (deobfuscated) file
+"""
+
+
+import puremagic
+import os
+
+# common extension from https://www.computerhope.com/issues/ch001789.htm
+commonextlist = [
+    "3g2",
+    "3gp",
+    "7z",
+    "ai",
+    "aif",
+    "apk",
+    "arj",
+    "asp",
+    "aspx",
+    "avi",
+    "bak",
+    "bat",
+    "bin",
+    "bin",
+    "bmp",
+    "c",
+    "cab",
+    "cda",
+    "cer",
+    "cfg",
+    "cfm",
+    "cgi",
+    "cgi",
+    "cgi",
+    "class",
+    "com",
+    "cpl",
+    "cpp",
+    "cs",
+    "css",
+    "csv",
+    "cur",
+    "dat",
+    "db",
+    "dbf",
+    "deb",
+    "dll",
+    "dmg",
+    "dmp",
+    "doc",
+    "docx",
+    "drv",
+    "email",
+    "eml",
+    "emlx",
+    "exe",
+    "flv",
+    "fnt",
+    "fon",
+    "gadget",
+    "gif",
+    "h",
+    "h264",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "ico",
+    "ini",
+    "iso",
+    "jar",
+    "java",
+    "jpeg",
+    "jpg",
+    "js",
+    "jsp",
+    "key",
+    "lnk",
+    "log",
+    "m4v",
+    "mdb",
+    "mid",
+    "midi",
+    "mkv",
+    "mov",
+    "mp3",
+    "mp4",
+    "mpa",
+    "mpeg",
+    "mpg",
+    "msg",
+    "msi",
+    "msi",
+    "odp",
+    "ods",
+    "odt",
+    "oft",
+    "ogg",
+    "ost",
+    "otf",
+    "part",
+    "pdf",
+    "php",
+    "php",
+    "pkg",
+    "pl",
+    "pl",
+    "pl",
+    "png",
+    "pps",
+    "ppt",
+    "pptx",
+    "ps",
+    "psd",
+    "pst",
+    "py",
+    "py",
+    "py",
+    "rar",
+    "rm",
+    "rpm",
+    "rss",
+    "rtf",
+    "sav",
+    "sh",
+    "sql",
+    "svg",
+    "swf",
+    "swift",
+    "sys",
+    "tar",
+    "tar",
+    "gz",
+    "tex",
+    "tif",
+    "tiff",
+    "tmp",
+    "toast",
+    "ttf",
+    "txt",
+    "vb",
+    "vcd",
+    "vcf",
+    "vob",
+    "wav",
+    "wma",
+    "wmv",
+    "wpd",
+    "wpl",
+    "wsf",
+    "xhtml",
+    "xls",
+    "xlsm",
+    "xlsx",
+    "xml",
+    "z",
+    "zip",
+]
+
+downloadextlist = [
+    "ass",
+    "avi",
+    "bat",
+    "bdmv",
+    "bin",
+    "bup",
+    "clpi",
+    "crx",
+    "db",
+    "diz",
+    "docx",
+    "epub",
+    "exe",
+    "flac",
+    "gif",
+    "gz",
+    "htm",
+    "html",
+    "icns",
+    "ico",
+    "idx",
+    "ifo",
+    "img",
+    "inf",
+    "info",
+    "ini",
+    "iso",
+    "jpg",
+    "log",
+    "m2ts",
+    "m3u",
+    "m4a",
+    "mkv",
+    "mp3",
+    "mp4",
+    "mpls",
+    "mx",
+    "nfo",
+    "nib",
+    "nzb",
+    "otf",
+    "par2",
+    "part",
+    "pdf",
+    "pem",
+    "php",
+    "plist",
+    "png",
+    "py",
+    "r00",
+    "r01",
+    "r02",
+    "r03",
+    "r04",
+    "r05",
+    "r06",
+    "r07",
+    "r08",
+    "r09",
+    "r10",
+    "r11",
+    "r12",
+    "r13",
+    "r14",
+    "r15",
+    "r16",
+    "r17",
+    "r18",
+    "r19",
+    "r21",
+    "r22",
+    "r23",
+    "r24",
+    "r25",
+    "r26",
+    "r27",
+    "r28",
```

Comment:
Feels a bit arbitrary to stop at `r28`. If you want to include these `.rXX` then I propose a specific check for `.rXX` and not all these separate extensions,

Reply:
Yes, please check my improved implementation